adapter/pkg/messaging/azure_listener.go

```diff
@@ -19,126 +19,49 @@
 package messaging
 
 import (
-	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
-	servicebus "github.com/Azure/azure-service-bus-go"
 	"context"
 	"strings"
-	"github.com/google/uuid"
+	"time"
+	servicebus "github.com/Azure/azure-service-bus-go"
+	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
 )
 
+func startBrokerConsumer(subscriptionMetaData Subscription, reconnectInterval time.Duration) {
+	var topicName = subscriptionMetaData.topicName
+	var subscriptionName = subscriptionMetaData.subscriptionName
 
-func startBrokerConsumer(topicName string, ns *servicebus.Namespace,
-	availableTopicList []*servicebus.TopicEntity, componentName string, opts ...servicebus.SubscriptionManagementOption) {
-
-	var topicExistForFurtherProcess bool
-	var subscriptionCreatedForFurtherProcess bool
-	parentContext := context.Background()
-	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Starting broker consumer for topic name : " +
-		topicName)
-
-	uniqueID := uuid.New()
-	//In Azure service bus subscription names can contain letters, numbers, periods (.), hyphens (-), and
-	// underscores (_), up to 50 characters. Subscription names are also case-insensitive.
-	var subscriptionName = componentName + "_" + uniqueID.String() + "_sub"
-
-	// TODO: (dnwick) Handle retry logic in error situations
-
-	if !isTopicExist(topicName, availableTopicList) {
-		//create the topic
-		topicManager := ns.NewTopicManager()
-		ctx, cancel := context.WithCancel(parentContext)
-		defer cancel()
-
-		_, err := topicManager.Put(ctx, topicName)
-		if err != nil {
-			logger.LoggerMgw.Error("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying to create " +
-				"topic " + topicName + " from azure service bus :%v", err)
-		} else {
-			logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Topic " + topicName + "created")
-			topicExistForFurtherProcess = true
-		}
-	} else {
-		topicExistForFurtherProcess = true
+	dataChannel := make(chan []byte)
+	if strings.EqualFold(topicName, notification) {
+		dataChannel = AzureNotificationChannel
+	} else if strings.EqualFold(topicName, tokenRevocation) {
+		dataChannel = AzureRevokedTokenChannel
 	}
+	parentContext := context.Background()
 
-	if topicExistForFurtherProcess {
-		logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Topic " + topicName + " exist. Can proceed")
-		subManager, err := ns.NewSubscriptionManager(topicName)
-
+	for {
+		//topic subscription client creation
+		topicSubscriptionClient, err := subscriptionMetaData.subscriptionManager.Topic.NewSubscription(subscriptionName)
```

Comment:
If something goes wrong in the connection, will the subscriptionManager still be valid? Or should we reset that?

Reply:
it will be valid .. I tested it